pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -211,14 +216,36 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	}
 
 	if strings.HasPrefix(fi.Name(), ".") {
-		glog.Errorf("Ignoring file: %s", fi.Name())
+		glog.Errorf("Ignoring file (starts with '.'): %s", fi.Name())
 		return nil
 	}
 
+	relative := strings.TrimPrefix(event.Name, w.path)
+	relative = strings.TrimPrefix(relative, "/")
+
 	if !fi.IsDir() {
+		if !strings.HasSuffix(fi.Name(), ".sock") {
+			glog.Errorf("Ignoring file (doesn't end with '.sock'): %s", fi.Name())
+			return nil
+		}
+
+		if strings.Count(relative, "/") > 1 {
+			glog.Errorf("Ignoring file (cannot be more than one level under the plugin directory): %s", fi.Name())
+			return nil
+		}
+
 		return w.handlePluginRegistration(event.Name)
 	}
 
+	if !strings.HasSuffix(relative, "/") {
+		relative += "/"
+	}
+
+	if strings.Count(relative, "/") > 1 {
```

Comment:
by reading the [PR](https://github.com/kubernetes/kubernetes/issues/69015), i believe filtering socket file is good enough to reduce the actual registration attempt, we will still see logs saying: non-socket files been ignored, we could increase log verbose level if think too many logs is a issue.

limit directory depth where plugin can stay has two drawbacks IMO:
1) it is hard to enforce, not sure everyone will read code/readme here.
2) it provides less flexibility to plugins: if they want to organize many plugins with directory structure .

Thanks